src/App.tsx

```diff
@@ -1,61 +1,128 @@
+import { FormEvent, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './components/types/User';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './components/types/Todo';
+
+const getUserById = (userId: number): User | null => {
+  const foundedUser = usersFromServer.find(user => user.id === userId);
+
+  return foundedUser || null;
+};
+
+const newTodoId = (todosArr: Todo[]): number => {
+  const todoIdArray = todosArr.map(todo => todo.id);
+  const newId = Math.max(...todoIdArray) + 1;
+
+  return newId;
+};
+
+const newTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>(newTodos);
+
+  const updatedTodo: Todo = {
+    id: newTodoId(todos),
+    userId: selectedUserId,
+    title: todoTitle,
+    completed: false,
+    user: getUserById(selectedUserId),
+  };
```

Comment:
Would transfer this constant to a block:
    if (todoTitle && selectedUserId !== 0) {
      // here
      setTodos([...todos, updatedTodo]);
      clearForm();
    }

Reply:
++